### PR TITLE
Development

### PR DESCRIPTION
**Pull request by balbazauras**

### CODE REVIEW

File: mysite/api/admin.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+from django.contrib import admin
+
+# Register your models here.
```
Comment (daiarn) [inline mysite/api/admin.py:3]: can remove this file if not used

File: mysite/api/views.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    sensor_id = request.query_params.get('sensor-id')
+    number_start = request.query_params.get('number-start')
+    number_end = request.query_params.get('number-end')
+    date_start = request.query_params.get('date-start')
+    date_end = request.query_params.get('date-end')
+    all_sensor_data = DataEntry.objects.all()
+    filtered_data = all_sensor_data.filter(value__gt=number_start, value__lt=number_end,
+                                           sensor__exact=sensor_id,
+                                           date__gt=date_start, date__lt=date_end
+                                           )
```
Comment (daiarn) [inline mysite/api/views.py:22]: you can just do:
```
filtered_data =  DataEntry.objects.filter(your filters here)
```

File: mysite/api/views.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    #   for item in data:
+    #   item.date = item.date.strftime('%Y-%m-%dT%H:%M:%S.%f%z')
```
Comment (daiarn) [inline mysite/api/views.py:24]: remove comments

File: mysite/mysite/settings.py [hunk 4]
```diff
@@ -29,10 +29,10 @@
 DEBUG = True
 
 STATICFILES_DIRS = (
-    os.path.join(BASE_DIR), ### ?????static?????????
+    os.path.join(BASE_DIR),  # ?????static?????????
```
Comment (daiarn) [inline mysite/mysite/settings.py:32]: want to remove that comments or still do not know what static files are?

File: mysite/mysite/settings.py [hunk 5]
```diff
@@ -79,6 +80,11 @@
     },
 ]
 
+REST_FRAMEWORK = {
+    'COERCE_DECIMAL_TO_STRING': False,
+
```
Comment (daiarn) [inline mysite/mysite/settings.py:85]: empty line not needed

File: mysite/mysite/settings.py [hunk 6]
```diff
@@ -94,7 +100,7 @@
         'PASSWORD': '87048bb7a9b593167d68209c5d2e898dd39adcafd25c7d501f6ca0fcad869040',
         'HOST': 'ec2-3-248-121-12.eu-west-1.compute.amazonaws.com',
         'PORT': '5432',
-        
+
```
Comment (daiarn) [inline mysite/mysite/settings.py:103]: empty line not needed

File: mysite/mysite/urls.py [hunk 7]
```diff
@@ -22,9 +22,9 @@
...
     path('admin/', admin.site.urls),
     path('', include('service.urls')),
-    path('api/',include(router.urls)),
-    re_path(r'^media/(?P<path>.*)$', serve,{'document_root':       settings.MEDIA_ROOT}), 
-    re_path(r'^static/(?P<path>.*)$', serve,{'document_root': settings.STATIC_ROOT}), 
+    path('api/', include('api.urls')),
+    re_path(r'^media/(?P<path>.*)$', serve,
+            {'document_root':       settings.MEDIA_ROOT}),
+    re_path(r'^static/(?P<path>.*)$', serve,
+            {'document_root': settings.STATIC_ROOT}),
```
Comment (daiarn) [inline mysite/mysite/urls.py:29]: what `re_path` do?

File: mysite/service/migrations/0008_alter_dataentry_value.py [hunk 8]
```diff
@@ -0,0 +1,18 @@
+# Generated by Django 4.0.4 on 2022-10-05 09:11
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
```
Comment (daiarn) [inline mysite/service/migrations/0008_alter_dataentry_value.py:6]: this migration is the same as `0010` you can migrate back to `0008` migration and delete `0009` and `0010` migrations

File: mysite/service/static/js/graph.js [hunk 9]
```diff
@@ -0,0 +1,65 @@
+function updateChart(sensor_id) {
+  request = {
+    "sensor-id": sensor_id,
+    "number-start": document.getElementById("number_start").value,
+    "number-end": document.getElementById("number_end").value,
+    "date-start": document.getElementById("date_start").value,
+    "date-end": document.getElementById("date_end").value,
+  };
+  $.get(container.data("posts-url"), request, function (data) {
+    console.log(data);
```
Comment (daiarn) [inline mysite/service/static/js/graph.js:10]: remove logging

File: mysite/service/static/js/graph.js [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+    var items = [];
```
Comment (daiarn) [inline mysite/service/static/js/graph.js:11]: you should not user `var` instead use `let` and `const` probably `const` here?

File: mysite/service/static/js/graph.js [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+    chartLabels.length = 0;
+    dataLabels.length = 0;
+    $.each(data, function (key, val) {
+      items.push("<li id='" + key + "'>" + val + "</li>");
```
Comment (daiarn) [inline mysite/service/static/js/graph.js:15]: that is not how you should build html elements. This the [tutorial](https://www.makeuseof.com/jquery-create-element/)

File: mysite/service/urls.py [hunk 12]
```diff
@@ -1,18 +1,18 @@
...
\ No newline at end of file
+    path('', views.home, name="home"),
+    path('register/', views.register_page, name="register"),
+    path('login/', views.login_page, name="login"),
+    path('logout/', views.logout_user, name="logout"),
+    path('system/<str:pk>/', views.system, name="system"),
+    path('sensor/<str:pk>/', views.sensor, name="sensor"),
+    path('create_system/<str:pk>/', views.create_system, name="create_system"),
+    path('update_system/<str:pk>/', views.update_system, name="update_system"),
+    path('delete_system/<str:pk>/', views.delete_system, name="delete_system"),
```
Comment (daiarn) [inline mysite/service/urls.py:16]: not sure if necessary but links shoud be with `-` so `create-system/` for example

File: mysite/service/views.py [hunk 13]
```diff
@@ -1,121 +1,128 @@
...
 def sensor(request, pk):
-    sensor = Sensor.objects.get(id=pk)
-    data = sensor.dataentry_set.all()
-    chartLabels = []
-    chartData = []
-    queryset = data.order_by('-date')[:20]
+    """Shows information about data entrys from single sensor """
+    # get a single sensor
+    single_sensor = Sensor.objects.get(id=pk)
+    # get all data entrys from this sensor
```
Comment (daiarn) [inline mysite/service/views.py:29]: comments here not really needed but you can keep it if you need them

File: mysite/service/views.py [hunk 14]
```diff
@@ -1,121 +1,128 @@
...
+    sensor_data = single_sensor.dataentry_set.all()
```
Comment (daiarn) [inline mysite/service/views.py:30]: you can add `.order_by('-date')` here

File: mysite/service/views.py [hunk 15]
```diff
@@ -1,121 +1,128 @@
...
+def create_system(request, pk):
+    """Creates single system """
+    technician = Technician.objects.get(id=pk)
+    form = SystemForm(initial={'technician': technician})
+    if request.method == 'POST':
+        form = SystemForm(request.POST)
         if form.is_valid():
             form.save()
-            print('Printing POST:',request.POST)
+            print('Printing POST:', request.POST)
```
Comment (daiarn) [inline mysite/service/views.py:65]: remove print

File: mysite/service/views.py [hunk 16]
```diff
@@ -1,121 +1,128 @@
...
+    """Updates single system data fields"""
+    single_system = System.objects.get(id=pk)
+    form = SystemForm(instance=single_system)
+
+    if request.method == 'POST':
+        form = SystemForm(request.POST, instance=system)
         if form.is_valid():
             form.save()
-            print('Printing POST:',request.POST)
+            print('Printing POST:', request.POST)
```
Comment (daiarn) [inline mysite/service/views.py:82]: remove print

**Review by daiarn — CHANGES_REQUESTED** [review]
Left few comments to fix

File: mysite/mysite/urls.py (hunk 7 above)
Comment (balbazauras) [reply mysite/mysite/urls.py:29]: These are not used anymore, followed tutorial which needed static files atm I dont need them

File: mysite/service/templates/service/system.html [hunk 17]
```diff
@@ -31,7 +31,7 @@ <h3>System location: {{system.location}}</h3>
         </table>
       </div>
       <div class="d-flex justify-content-center">
-        <a href="{% url '"
+        <a
```
Comment (daiarn) [inline mysite/service/templates/service/system.html:34]: is this ok?

**Review by daiarn — CHANGES_REQUESTED** [review]
ONe question and please remove cache files from commits with git ignore

**Pull request closed without merging**